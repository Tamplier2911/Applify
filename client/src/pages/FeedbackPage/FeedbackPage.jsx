// import "./FeedbackPage.scss";
import React, { useEffect } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  selectIsLoading,
  selectCurrentlyLoadedFeeds,
} from "../../redux/feedbacks/feedbacks.selectors";
import { loadFeedbacksStart } from "../../redux/feedbacks/feedbacks.actions";

// components
import FeedbacksCollection from "../../components/FeedbacksCollection/FeedbacksCollection";
import WithSpinnerHOC from "../../components/WithSpinnerHOC/WithSpinnerHOC";
// import FeedbackCreate from "../../components/FeedbackCreate/FeedbackCreate";

// JS Rendering CSS
import { FeedbackPageContainer } from "./FeedbackPageStyles";

// FeedbacksCollection Buffed With Spinner
const FeedbacksCollectionWithSpinner = WithSpinnerHOC(FeedbacksCollection);

const FeedbackPage = ({ loadFeedbacksStart, loading, allFeeds }) => {
  useEffect(() => {
    if (!allFeeds.length) {
      loadFeedbacksStart();
    }
  }, [loadFeedbacksStart, allFeeds.length]);

  return (
    <FeedbackPageContainer>
      <FeedbacksCollectionWithSpinner
        isLoading={loading}
        feedbacks={allFeeds}
      />
      {/* <FeedbackCreate method="POST" /> */}
    </FeedbackPageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  loading: selectIsLoading,
  allFeeds: selectCurrentlyLoadedFeeds,
});

export default connect(mapStateToProps, {
  loadFeedbacksStart,
})(FeedbackPage);
