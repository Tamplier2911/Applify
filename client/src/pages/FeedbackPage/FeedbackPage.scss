.feedback-page {
  display: grid;
  grid-row-gap: 5rem;

  // @media only screen and (max-width: 980px) {
  //   grid-template-rows: 68vh 1fr;
  //   grid-row-gap: 0rem;
  // }

  // @media only screen and (max-width: 720px) {
  //   grid-template-rows: 60vh 1fr;
  //   grid-row-gap: 7rem;
  // }

  // @media only screen and (max-width: 540px) {
  //   margin-top: 2rem;
  //   grid-template-rows: 80vh 1fr;
  // }

  @media only screen and (max-width: 768px) {
    grid-row-gap: 5rem;
  }

  @media only screen and (max-width: 425px) {
    grid-row-gap: 5rem;
  }
}
