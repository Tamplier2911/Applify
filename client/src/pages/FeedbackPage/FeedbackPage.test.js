import React from "react";
import toJson from "enzyme-to-json";

// component
import FeedbackPage from "./FeedbackPage";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<FeedbackPage />", () => {
  const initialStore = {
    feedbacks: {
      allFeedbacks: [],
      currentlyLoaded: [],
      totalLoaded: 0,
      isLoading: false,
      errorMessage: null,
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<FeedbackPage store={store} />);
  });

  it("renders without error", () => {
    const feedbackPage = findByTestAttr(wrapper, "feedback-page");
    expect(feedbackPage.length).toBe(1);
  });

  it("renders children - feedback-page-with-spinner", () => {
    const feedbackPageWithSpinner = findByTestAttr(
      wrapper,
      "feedback-page-with-spinner"
    );
    expect(feedbackPageWithSpinner.length).toBe(1);
  });

  it("matches snapshot", () => {
    const feedbackPage = findByTestAttr(wrapper, "feedback-page");
    expect(toJson(feedbackPage)).toMatchSnapshot();
  });
});
