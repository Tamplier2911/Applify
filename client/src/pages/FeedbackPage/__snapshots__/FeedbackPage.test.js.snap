// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackPage /> matches snapshot 1`] = `
<styled.section
  data-test="feedback-page"
>
  <Component
    data-test="feedback-page-with-spinner"
    feedbacks={Array []}
    isLoading={false}
  />
</styled.section>
`;
